src/blog/emscripten-standalone-wasm.md

```diff
@@ -0,0 +1,180 @@
+
+Okay, so far so good: Emscripten can either emit JavaScript + WebAssembly as it always did, and now it can also emit just WebAssembly by itself, which lets you run it in places that don't have JavaScript like wasm runtimes, or you can write your own custom JavaScript loader code, etc. Now let's talk about the background and the technical details!
+
+## WebAssembly's two standard APIs
+
+WebAssembly can only access the APIs it receives as imports - the core wasm spec has no concrete API details. Given the current trajectory of wasm, it looks like there will be 3 main categories of APIs that people import and use:
+
+- **Web APIs**: This is what wasm programs use on the Web, which are the existing standardized APIs that JavaScript can use too. Currently these are called indirectly, through JS glue code, but in the future with [interface types](https://github.com/WebAssembly/interface-types/blob/master/proposals/interface-types/Explainer.md) they will be called directly.
+- **WASI APIs**: WASI focuses on standardizing APIs for wasm on the server.
+- **Other APIs**: Various custom embeddings will define their own application-specific APIs. For example, we gave the example earlier of an image editor with wasm plugins that implement an API to do visual effects. Note that a plugin might also have access to “system” APIs, like a native dynamic library would, or it might be very sandboxed and have no imports at all (if the embedding just calls its methods).
+
+WebAssembly is in the interesting position of having [two standardized sets of APIs](https://www.goodreads.com/quotes/589703-the-good-thing-about-standards-is-that-there-are-so). This does makes sense in that one is for the Web and one for the server, and those environments do have different requirements; for similar reasons Node.js does not have identical APIs to JavaScript on the Web.
+
+However, there is more than the Web and the server, in particular there are also wasm plugins. For one thing, plugins can run inside an application that may be on the Web (just like [JS plugins](https://www.figma.com/blog/an-update-on-plugin-security/#a-technology-change)) or off the Web; for another, regardless of where the embedding application is, a plugin environment is not a Web nor a server environment. So it's not immediately obvious which sets of APIs will be used - it may depend on the code being ported, the wasm runtime being embedded, etc.
```

Comment:
Should this be a prelude to the plugins section? There seems to be some intersection between this paragraph and the first one there.

Reply:
It is about plugins, so I see your point. But I think it makes sense here since it's about APIs in plugins, and it concludes the API section basically: we have 2 standard APIs, which sort of makes sense maybe, but it's a complicated picture in particular due to plugins.

Reply:
I guess this is fine as-is.